airbyte-cdk/python/unit_tests/sources/file_based/in_memory_files_source.py

```diff
@@ -42,48 +43,61 @@ def __init__(
         file_write_options: Mapping[str, Any],
         max_history_size: int,
     ):
+        # Attributes required for test purposes
+        self.files = files
+        self.file_type = file_type
```

Comment:
will the S3 source have the same restriction of one `file_type` per source?

I assume not, but it's not clear who owns the `file_type` now that it isn't part of the `RemoteFile` anymore

Reply:
`file_type` is defined per-stream.